core/templates/dev/head/domain/collection/CollectionRightsBackendApiService.js

```diff
@@ -67,6 +150,22 @@ oppia.factory('CollectionRightsBackendApiService', [
             _setCollectionStatus(
               collectionId, collectionVersion, false, resolve, reject);
           });
+        },
+
+        /**
+         * Returns true if the collection is private given the collectionId.
+         * This method will check the collection rights object exists, if so,
+         * return the result from the cache. However, this will not cache the
+         * collection rights object if it does not exist in the cache.
+         */
+        isCollectionPrivate: function(collectionId) {
```

Comment:
Ah yes, that's an issue. I'm surprised that promises don't work. An alternative is to make the method non-async and to have the caller retry after 1-2 sec if a null response is obtained. But mixing sync and async in one method seems to me like a recipe for confusion...